### PR TITLE
Generalize AsymmetricUnifiedFocalLoss for multi-class and align interface

### LINKED ISSUE #8603: Add sigmoid/softmax interface for `AsymmetricUnifiedFocalLoss`
**Is your feature request related to a problem? Please describe.**
It seems that `AsymmetricUnifiedFocalLoss` has not yet been updated to the latest interface like `FocalLoss`. It is a general loss function and we should make it easier for everyone to use. We can also allow multiple classes segmentation since it only supports binary one accordingly

**Describe the solution you'd like**
Follow the interface of `FocalLoss`. Also, we don't need to reinvent the wheel. I am sure we can use the available loss function to replace the [implementation](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/unified_focal_loss.py#L189).

**Describe alternatives you've considered**
No

**Additional context**
No

### PR DESCRIPTION
**Pull request by ytl0623**
Fixes #8603  .

### Description

Addresses the limitations of `AsymmetricUnifiedFocalLoss` and its dependencies (`AsymmetricFocalLoss`, `AsymmetricFocalTverskyLoss`). The previous implementations were hard-coded for binary segmentation and lacked the standard `sigmoid`/`softmax`/`to_onehot_y` interface common to other MONAI losses.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ytl0623** [comment]
Hi @KumoLiu, @Nic-Ma  and @ericspod,

Sorry to bother.
Could you please review the PR or give some tips?

Thanks in advance!

File: monai/losses/unified_focal_loss.py [hunk 1]
```diff
@@ -23,218 +23,272 @@
...
     Reimplementation of the Asymmetric Focal Tversky Loss described in:
 
     - "Unified Focal Loss: Generalising Dice and Cross Entropy-based Losses to Handle Class Imbalanced Medical Image Segmentation",
-    Michael Yeung, Computerized Medical Imaging and Graphics
+      Michael Yeung, Computerized Medical Imaging and Graphics
     """
 
     def __init__(
         self,
-        to_onehot_y: bool = False,
```
Comment (ericspod) [inline monai/losses/unified_focal_loss.py:37]: Why is `to_onehot_y` being removed? I think users will still want this functionality so I would leave it in and add `include_background` as a new last argument. Even if we do want to remove it we need to use the [deprecation decorators to mark the argument removed](https://github.com/Project-MONAI/MONAI/blob/69f3dd26ed2a65e89ae89d951bb16f2dcb4d7c5d/monai/utils/deprecate_utils.py#L123) but still keep it for a version or two.

**Review by ericspod — COMMENTED** [review]
Hi @ytl0623 thanks for the contribution. I had only a minor comment but please look at the greptile comments. A few look spurious they should all be resolved.
